source/dub/recipe/packagerecipe.d

```diff
@@ -171,50 +213,139 @@ struct ConfigurationInfo {
+
+        if (p.node.nodeID == NodeID.scalar)
+            return YAMLDep(Dependency(p.node.as!string));
+        auto d = p.parseField!"representation";
+        if (d.path.length)
+        {
+            auto r = YAMLDep(Dependency(NativePath(d.path)));
+            r.optional = d.optional;
+            r.default_ = d.default_;
+            return r;
+        }
+        else if (d.repository.length)
+        {
+            auto r = YAMLDep(Dependency(Repository(d.repository, d.version_)));
+            r.optional = d.optional;
+            r.default_ = d.default_;
+            return r;
+        }
+        else
+        {
+            assert(d.version_.length);
+            auto r = YAMLDep(Dependency(VersionRange.fromString(d.version_)));
+            r.optional = d.optional;
+            r.default_ = d.default_;
+            return r;
+        }
+    }
+}
+
 /// This keeps general information about how to build a package.
 /// It contains functions to create a specific BuildSetting, targeted at
 /// a certain BuildPlatform.
 struct BuildSettingsTemplate {
-	Dependency[string] dependencies;
-	BuildSettingsTemplate[string] dependencyBuildSettings;
-	string systemDependencies;
-	TargetType targetType = TargetType.autodetect;
-	string targetPath;
-	string targetName;
-	string workingDirectory;
-	string mainSourceFile;
-	string[string] subConfigurations;
-	string[][string] dflags;
-	string[][string] lflags;
-	string[][string] libs;
-	string[][string] sourceFiles;
-	string[][string] sourcePaths;
-	string[][string] excludedSourceFiles;
-	string[][string] injectSourceFiles;
-	string[][string] copyFiles;
-	string[][string] extraDependencyFiles;
-	string[][string] versions;
-	string[][string] debugVersions;
-	string[][string] versionFilters;
-	string[][string] debugVersionFilters;
-	string[][string] importPaths;
-	string[][string] stringImportPaths;
-	string[][string] preGenerateCommands;
-	string[][string] postGenerateCommands;
-	string[][string] preBuildCommands;
-	string[][string] postBuildCommands;
-	string[][string] preRunCommands;
-	string[][string] postRunCommands;
-	string[string][string] environments;
-	string[string][string] buildEnvironments;
-	string[string][string] runEnvironments;
-	string[string][string] preGenerateEnvironments;
-	string[string][string] postGenerateEnvironments;
-	string[string][string] preBuildEnvironments;
-	string[string][string] postBuildEnvironments;
-	string[string][string] preRunEnvironments;
-	string[string][string] postRunEnvironments;
+	@Optional YAMLDep[string] dependencies;
+	@Optional BuildSettingsTemplate[string] dependencyBuildSettings;
+	@Optional string systemDependencies;
+	@Optional TargetType targetType = TargetType.autodetect;
+	@Optional string targetPath;
+	@Optional string targetName;
+	@Optional string workingDirectory;
+	@Optional string mainSourceFile;
+	@Optional string[string] subConfigurations;
+	@Name("dflags", true) string[][string] dflags;
```

Comment:
shouldn't all of these starting here and below be optional as well?

+ same as above with a new `@StartsWith` UDA.

Reply:
Actually I can remove `@Optional` on the one above. Because `@Optional` is transitive, and was applied to the field that is `alias this`-ed, they are already considered optional.

Reply:
Actually I can't, because they are also member of an AA. So `@Optional` is correct.